tests/Kernel/CorporateBlocks/CorporateBlocksTestBase.php

```diff
@@ -5,19 +5,161 @@
+
+    $graphs = [
+      'corporate_body' => 'http://publications.europa.eu/resource/authority/corporate-body',
+    ];
+    $this->container->get('rdf_skos.skos_graph_configurator')->addGraphs($graphs);
+
+    $this->configFactory = $this->container->get('config.factory');
+    $corporate_site_info = $this->configFactory->getEditable('oe_corporate_site_info.settings');
+    $corporate_site_info->setData([
+      'site_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/ACP-EU_JA',
+      'content_owners' => ['http://publications.europa.eu/resource/authority/corporate-body/COMMU'],
+    ]);
+    $corporate_site_info->save();
+
+    $config = $this->configFactory->getEditable('system.site');
+    $config->set('name', 'OpenEuropa');
+    $config->save();
+  }
+
+  /**
+   * Render a corporate footer block with test data.
+   *
+   * @param string $type
+   *   The type of block, ec or eu.
+   * @param array $test_data
+   *   The test data for config and assertion.
+   *
+   * @return string
+   *   The rendered HTML.
+   */
+  protected function renderCorporateBlocksFooter(string $type, array &$test_data): string {
+    // Override corporate block footer config with test data.
+    $config_name = "oe_corporate_blocks.{$type}_data.footer";
+    $fixture_name = "{$type}_footer.yml";
+    $block_id = "oe_corporate_blocks_{$type}_footer";
+
+    /* @var $config_obj \Drupal\Core\Config\Config */
+    $config_obj = $this->configFactory->getEditable($config_name);
+    $test_data = $this->getFixtureContent($fixture_name);
+    $config_obj->setData($test_data);
+    $config_obj->save();
+
+    // Setup and render footer block.
+    $config = [
+      'id' => $block_id,
+      'label' => 'OpenEuropa footer block',
+      'provider' => 'oe_corporate_blocks',
+      'label_display' => '0',
+    ];
+    $build = $this->buildBlock($block_id, $config);
+
+    return $this->renderRoot($build);
+  }
+
+  /**
+   * We assigned each section two links, assert that content and order.
+   *
+   * @param \Symfony\Component\DomCrawler\Crawler $section
+   *   The footer section.
+   * @param string $branding
+   *   Ecl branding, core/standardised.
+   * @param array $expected
+   *   The expected data.
+   */
+  protected function assertLinkList(Crawler $section, string $branding, array $expected): void {
+    $actual = $section->filter('ul li:nth-child(1) > a');
+    $this->assertEquals($expected['0']['href'], $actual->attr('href'));
+    $this->assertEquals($expected['0']['label'], $actual->text());
+    $this->assertEquals("ecl-link ecl-link--standalone ecl-footer-{$branding}__link", $actual->attr('class'));
+
+    $actual = $section->filter('ul li:nth-child(2) > a');
+    $this->assertEquals($expected['1']['href'], $actual->attr('href'));
+    $this->assertEquals($expected['1']['label'], $actual->text());
+    $this->assertEquals("ecl-link ecl-link--standalone ecl-footer-{$branding}__link", $actual->attr('class'));
```

Comment:
This function only works if we assign to each section two links, so what if we want to test that, if we remove a link from config, cache is invalidated and we now show 1 link only?

Each method/function shouldn't be aware of anything regarding its caller, let's keep this as a general principle. If you want to have a link assertion method, make sure that it asserts only one link, so that the caller can have full control.